File: demos/ios/Demos/Demos/views/GroupBarChart2.m
//
//  MultiIntervalUIView2.m
//  F2Native
//
//  Created by 青原 on 2020/12/22.
//  Copyright © 2020 com.alipay.xgraph. All rights reserved.
//

#import "GroupBarChart2.h"

@implementation GroupBarChart2

- (void)chartRender {
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_singleIntervalChart_under_zero" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    self.chart.canvas(self.canvasView).padding(20, 10, 20, 0);
    self.chart.source([F2Utils toJsonArray:jsonData]);
    self.chart.axis(@"tem", @{@"grid": @{@"stroke": @"#000"}});
    self.chart.interval().position(@"time*tem").color(@"city", @[]).adjust(@"dodge");
    self.chart.scale(@"time", @{@"range": @[@(0.1), @(0.9)]});
    self.chart.scale(@"tem", @{@"nice": @(YES)});
    self.chart.tooltip(@{});
    self.chart.render();
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/GroupBarChart2.swift
//
//  GroupBarChart2.swift
//  F2Native
//
//  Created by Westin on 2024-06-09.
//

import F2

class GroupBarChart2: BaseLineChart {
    
    override func chartRender() {
        guard let jsonPath = Bundle.main.path(forResource: "Res/mockData_singleIntervalChart_under_zero", ofType: "json"),
              let jsonDataString = try? String(contentsOfFile: jsonPath),
              let canvasView = self.canvasView,
              let chart = self.chart else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonDataString)
        chart.canvas()(canvasView).padding()(20, 10, 20, 0)
        chart.source()(jsonData)
        chart.axis()("tem", ["grid": ["stroke": "#000"]])
        chart.interval()().position()("time*tem").color()("city", []).adjust()("dodge")
        chart.scale()("time", ["range": [0.1, 0.9]])
        chart.scale()("tem", ["nice": true])
        chart.tooltip()([:])
        chart.animate()(true)
        chart.render()()
    }
}


